File: samples/src/main/kotlin/dev/pubgkt/PlayersSample.kt
package dev.pubgkt

import dev.pubgkt.players.getPlayersByNames
import dev.pubgkt.stats.season.getSeasonStatsByGameModeAndPlayers

suspend fun main(args: Array<String>) {
    val apiKey = args.getOrNull(0) ?: error("Usage: PlayersSample <api-key> [accountId] [playerName]")
    val accountId = args.getOrElse(1) { "account.91186dcca3cb4ad198fac1e4ab1d5b80" }
    val playerName = args.getOrElse(2) { "sparkingg" }
    val clanId = args.getOrElse(3) { "clan.d52aad6adcfb4c4783a85eb250f6e822" }

    val api = PubgApi(apiKey)
    val playerIds = api.getPlayersByNames("sparkingg", "TGLTN").map { it.id }
    println(
        api.getSeasonStatsByGameModeAndPlayers(
            seasonId = "division.bro.official.pc-2018-41",
            gameMode = GameMode.SQUAD_FPP,
            accountIds = playerIds,
        ),
    )

//    println("=== getPlayerByAccountId ===")
//    val player = api.getPlayerByAccountId(accountId)
//    println(player)
//
//    println("\n=== getPlayersById ===")
//    val byIds = api.getPlayersById(accountId)
//    byIds.forEach(::println)
//
//    println("\n=== getPlayersByNames ===")
//    val byNames = api.getPlayersByNames(playerName)
//    byNames.forEach(::println)
//
//    println("\n=== getClanById ===")
//    val clan = api.getClanById(clanId)
//    println(clan)
}

File: samples/swift/src/PlayersSample.swift
import Foundation
import PubgKt

func main() async throws {
    guard CommandLine.arguments.count > 1 else {
        print("Usage: PlayersSample <api-key>")
        return
    }
    let apiKey = CommandLine.arguments[1]
    let api = PubgApi(apiKey: apiKey, rateLimiter: RateLimiterCompanion.shared.None, retry: NoRetry.shared)

    print("=== getPlayersByNames ===")
    let byNames = try await api.getPlayersByNames(playerNames: ["sparkingg", "TGLTN"], platform: Platform.steam)
    for player in byNames { print(player) }

    print("\n=== getPlayerByAccountId ===")
    let accountId = byNames.first!.id
    let player = try await api.getPlayerByAccountId(accountId: accountId, platform: Platform.steam)
    print(player)

    print("\n=== getPlayersById ===")
    let ids = byNames.map { $0.id }
    let byIds = try await api.getPlayersById(accountIds: ids, platform: Platform.steam)
    for player in byIds { print(player) }
}

Task {
    do {
        try await main()
    } catch {
        print("Error: \(error)")
    }
    exit(0)
}
dispatchMain()
